xarray/core/indexing.py

```diff
@@ -72,6 +73,42 @@ def _expand_slice(slice_, size):
     return np.arange(*slice_.indices(size))
 
 
+def maybe_convert_to_slice(indexer, size):
```

Comment:
This was a fun utility function to write -- too bad we don't need it anymore! :)